craft_providers/base.py

```diff
@@ -786,6 +792,63 @@ def _post_setup_network(self, executor: Executor) -> None:
         """
         return
 
+    def _mount_shared_cache_dirs(self, executor: Executor) -> None:
+        """Mount a shared cache directories for this base.
+
+        e.g.
+            pip cache (normally $HOME/.cache/pip)
+
+        This will only be run if caching is enabled for this instance.
+
+        This step should usually be extended, but may be overridden if common
+        cache directories are in unusual locations.
+        """
+        if self._cache_path is None:
+            logger.debug("No cache path set, not mounting cache directories.")
+            return
+
+        # Get the real path with additional tags.
+        host_base_cache_path = pathlib.Path(
+            os.path.realpath(self._cache_path), self.compatibility_tag, str(self.alias)
+        )
+
+        try:
+            host_base_cache_path.mkdir(parents=True, exist_ok=True)
+            host_base_cache_path.chmod(0o777)
```

Comment:
I don't think we need this for most caches — lxd and multipass should do the account mapping for us.

Reply:
Just tested, inside the lxc it will not get mapped.

```
root@craft-buildd-instance:~# cd .cache/
root@craft-buildd-instance:~/.cache# ls -la
total 10
drwxr-xr-x 3 root   root    3 Sep 21 08:15 .
drwx------ 5 root   root    7 Sep 21 08:15 ..
drwxrwxrwx 2 nobody nogroup 2 Sep 21 08:15 pip
root@craft-buildd-instance:~/.cache# 
```